pygfx/renderers/wgpu/wgsl/text.wgsl

```diff
@@ -227,6 +227,16 @@ fn fs_main(varyings: Varyings) -> FragmentOutput {
     apply_clipping_planes(varyings.world_pos);
     var out = get_fragment_output(varyings.position.z, color_out);
 
+    // Move text closer to camera, since its often overlaid on something.
+    // The text is moved closer than the outline so that the outline of one character
+    // does not hide the text of an other.
+    // The depth buffer is a 32 bit float.
+    // Setting the incremented depth to be less than 1e-6 caused
+    // z-fighting to occur leading to artifacts.
+    // See: https://github.com/pygfx/pygfx/pull/776#issuecomment-2149374275
+    // And: https://github.com/pygfx/pygfx/pull/774#issuecomment-2147458827
+    out.depth = varyings.position.z - 1e-6 * (2.0 - outline);
```

Comment:
I suspect that at the edge of the outline (where it moves from 0-1) there is z-fighting going on. This (I think) would solve that.
```suggestion
    out.depth = varyings.position.z - seslect(1e-6 , 2e-6 , outline == 0.0);
```

Reply:
This didn't really help. I might have to sleep on this. feel free to push.